omegaconf/_utils.py

```diff
@@ -501,21 +467,18 @@ def is_primitive_type(type_: Any) -> bool:
 
 def _is_interpolation(v: Any) -> bool:
     if isinstance(v, str):
-        ret = get_value_kind(v) in (
-            ValueKind.INTERPOLATION,
-            ValueKind.STR_INTERPOLATION,
-        )
-        assert isinstance(ret, bool)
-        return ret
+        return bool(get_value_kind(v) == ValueKind.INTERPOLATION)
```

Comment:
I think I prefer the assert.
this will happily convert NotImplemented (which is a possible return value from _eq_) into a bool.

Reply:
I don't think `==` can return `NotImplemented` (`__eq__()` can, but not the `==` operator).

That being said, https://docs.python.org/3/library/operator.html says that "Note that these functions can return any value, which may or may not be interpretable as a Boolean value". Then it refers to a page (https://docs.python.org/3/reference/expressions.html#comparisons) that says "Comparisons yield boolean values: True or False".

I'll revert it to assert just to be safe.

Reply:
> I'll revert it to assert just to be safe.

Done in 83e2c8c